### PR TITLE
Fix JSON_QUERY Function parse error while path is all number

### PR DESCRIPTION
**Pull request by KevinyhZou**
### Changelog category (leave one):
- Bug Fix 

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix the JSON_QUERY function can not parse the json string while path is numberic. like in the query SELECT JSON_QUERY('{"123":"abcd"}', '$.123'), we would encounter the exceptions
```
DB::Exception: Unable to parse JSONPath: While processing JSON_QUERY('{"123":"acd"}', '$.123'). (BAD_ARGUMENTS)
```

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp [hunk 1]
```diff
@@ -51,13 +51,18 @@ bool ParserJSONPathMemberAccess::parseImpl(Pos & pos, ASTPtr & node, Expected &
         }
         const auto * last_begin = *pos->begin == '.' ? pos->begin + 1 : pos->begin;
         const auto * last_end = pos->end;
+        const String numberic_member_name = String(last_begin, last_end);
```
Comment (taiyang-li) [inline src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp:54]: const String numberic_member_name(last_begin, last_end)

**Review by taiyang-li — CHANGES_REQUESTED** [review]

File: src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp (hunk 1 above)
Comment (KevinyhZou) [reply src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp]: Fixed

**Comment by robot-ch-test-poll3** [comment]
This is an automated comment for commit f75b5c91b70bd3cd4b64e8ba79b6c9de330e2661 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/53470/f75b5c91b70bd3cd4b64e8ba79b6c9de330e2661/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp [hunk 2]
```diff
@@ -51,13 +51,18 @@ bool ParserJSONPathMemberAccess::parseImpl(Pos & pos, ASTPtr & node, Expected &
         }
         const auto * last_begin = *pos->begin == '.' ? pos->begin + 1 : pos->begin;
         const auto * last_end = pos->end;
+        const String numberic_member_name(last_begin, last_end);
```
Comment (lgbo-ustc) [inline src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp:54]: move this into the `else if` block, maybe better

File: tests/queries/0_stateless/01889_sql_json_functions.reference [hunk 3]
```diff
@@ -77,6 +77,8 @@ SELECT JSON_QUERY('{"array":[[0, 1, 2, 3, 4, 5], [0, -1, -2, -3, -4, -5]]}', '$.
 [0, 1, 4, 0, -1, -4]
 SELECT JSON_QUERY('{"1key":1}', '$.1key');
 [1]
+SELECT JSON_QUERY('{"123":1}', '$.123');
+[1]
```
Comment (lgbo-ustc) [inline tests/queries/0_stateless/01889_sql_json_functions.reference:81]: add some tests like
```sql
SELECT JSON_QUERY('{"123":1}', '$[123]');
```
Not sure whether is this a valid path.

File: src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp (hunk 2 above)
Comment (KevinyhZou) [reply src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp]: done

File: tests/queries/0_stateless/01889_sql_json_functions.reference (hunk 3 above)
Comment (KevinyhZou) [reply tests/queries/0_stateless/01889_sql_json_functions.reference:81]: done

**Comment by KevinyhZou** [comment]
cc @Avogar

**Review by Avogar — APPROVED** [review]

**Pull request merged**